pandas/core/indexes/base.py

```diff
@@ -2385,8 +2393,18 @@ def intersection(self, other):
             indexer = indexer[indexer != -1]
 
         taken = other.take(indexer)
+
+        if sort:
```

Comment:
ok, I still think we can use ``_get_reconcile_name_object`` here (maybe need to wrap ``taken`` in ``Index``), but can do in a followup.